packages/flutter/lib/src/painting/continuous_rectangle_border.dart

```diff
@@ -3,58 +3,251 @@
+///   precisely quarter circles.
+/// * [ContinuousStadiumBorder], which is a stadium whose two edges have a
+///   continuous transition into its two 180º curves.
+/// * [StadiumBorder], which is a rectangle with semi-circles on two parallel
+///   edges.
 class ContinuousRectangleBorder extends ShapeBorder {
-  /// The arguments must not be null.
+  /// Creates a continuous cornered rectangle border.
+  ///
+  /// The [cornerRadius] argument must not be null.
   const ContinuousRectangleBorder({
-    this.side = BorderSide.none,
-    this.borderRadius = BorderRadius.zero,
-  }) : assert(side != null),
-       assert(borderRadius != null);
+    this.cornerRadius = 0.0,
+  }) : assert(cornerRadius != null);
 
   /// The radius for each corner.
   ///
-  /// Negative radius values are clamped to 0.0 by [getInnerPath] and
-  /// [getOuterPath].
-  final BorderRadiusGeometry borderRadius;
+  /// The radius will be clamped to 0 if a value less than 0 is entered as the
+  /// radius. By default the radius is 0. This value must not be null.
+  ///
+  /// Unlike [RoundedRectangleBorder], there is only a single radius value used
+  /// to describe the radius for every corner.
+  final double cornerRadius;
+
+  Path _getPath(Rect rect) {
+    double limitedRadius;
+    final double width = rect.width;
+    final double height = rect.height;
+    final double centerX = rect.center.dx;
+    final double centerY = rect.center.dy;
+    final double radius = math.max(0.0, cornerRadius);
+    final double minSideLength = math.min(rect.width, rect.height);
+
+    // These equations give the x and y values for each of the 8 mid and corner
+    // points on a rectangle.
+    //
+    // For example, leftX(k) will give the x value on the left side of the shape
+    // that is precisely `k` distance from the left edge of the shape for the
+    // predetermined 'limitedRadius' value.
+    double leftX(double x) { return centerX + x * limitedRadius - width / 2; }
+    double rightX(double x) { return centerX - x * limitedRadius + width / 2; }
+    double topY(double y) { return centerY + y * limitedRadius - height / 2; }
+    double bottomY(double y) { return centerY - y * limitedRadius + height / 2; }
+
+    // Renders the default superelliptical rounded rect shape where there are
+    // 4 straight edges and 4 90º corners. Approximately renders a superellipse
+    // with an n value of 5.
+    //
+    // Paths and equations were inspired from the code listed on this website:
+    // https://www.paintcodeapp.com/news/code-for-ios-7-rounded-rectangles
+    //
+    // The shape is drawn from the top midpoint to the upper right hand corner
+    // in a clockwise fashion around to the upper left hand corner.
+    Path bezierRoundedRect () {
+      return Path()
+        ..moveTo(leftX(1.52866483), topY(0.0))
+        ..lineTo(rightX(1.52866471), topY(0.0))
+        ..cubicTo(rightX(1.08849323), topY(0.0),
```

Comment:
Skia team had a discussion about this "squircles" about 1.5 years ago. It seems that Skia wanted to add something like `bool SkPath::asSuperellipse(SkRect*, int* exp);` but it's not there yet.

From my own CPU path rasterization experience, cubics are more expensive than quadratics. Although it's unlikely to be a bottleneck for the whole app, it might be safer to just use quadratics as I heard that you can either use a single cubic or 2 quadratics to approximate the squircle.

CC Skia's GPU guru @bsalomon @csmartdalton here for comments about cubics vs. quadratics performance on GPU.

Of course, it would be better if Skia just added `bool SkPath::asSuperellipse(SkRect*, int* exp);` API so it can freely change the most performant implementation without Flutter modifying the source code. Not sure if the benefit is worth the maintenance cost (CC @reed-at-google ).

Reply:
The shapes are not yet linked as this PR needs to land before the shape images can be generated. Tho you should be able to see the shapes at this link: https://github.com/flutter/assets-for-api-docs/pull/55/files